### PR TITLE
user-record: don't synthesize default list of self-modfiable fields for non-regular users.

### PR DESCRIPTION
**Pull request by poettering**
A follow-up for a192250eda1e5cc1f8fc799cf9b85d37e7fa0519

/cc @AdrianVovk

### CODE REVIEW

File: src/shared/user-record-show.c [hunk 1]
```diff
@@ -28,14 +28,19 @@ const char* user_record_state_color(const char *state) {
...
+                const char **value) {
+
         assert(heading);
 
         /* Helper function for printing the various self_modifiable_* fields from the user record */
 
         if (strv_isempty((char**) value))
                 /* Case 1: the array is explicitly set to be empty by the administrator */
-                printf("%13s %sDisabled by Administrator%s\n", heading, ansi_highlight_red(), ansi_normal());
+                printf("%13s %sdisabled by Administrator%s\n", heading, user_record_disposition(hr) == USER_REGULAR ? ansi_highlight_red() : ansi_normal(), ansi_normal());
```
Comment (AdrianVovk) [inline src/shared/user-record-show.c]: Hmm, well in this case the setting wasn't disabled by the admin.

Really there's probably 4 cases:

- Default regular: field is null but value isn't empty
- Default other: field is null and value is empty
- Disabled by admin: field is nonnull but not empty
- Custom list: field is nonempty

**Comment by poettering** [comment]
Pushed an update. PTAL, @AdrianVovk

File: src/shared/user-record.c [hunk 2]
```diff
@@ -2165,8 +2165,15 @@ const char** user_record_self_modifiable_fields(UserRecord *h) {
 
         assert(h);
 
+        /* Note: if the self_modifiable_fields field in user-record is NULL we'll apply a default, if we have
+         * one. If it is a non-NULL empty strv, we'll report it as explicit empty list. When we the field is
```
Comment (keszybz) [inline src/shared/user-record.c]: "we the field"?

**Review by keszybz — APPROVED** [review]

File: src/shared/user-record-show.c [hunk 3]
```diff
@@ -28,14 +28,21 @@ const char* user_record_state_color(const char *state) {
...
-                /* Case 1: the array is explicitly set to be empty by the administrator */
-                printf("%13s %sDisabled by Administrator%s\n", heading, ansi_highlight_red(), ansi_normal());
+        if (!value)
+                /* Case 1: no value is set and no default either */
+                printf("%13s %snone%s\n", heading, ansi_highlight(), ansi_normal());
+        else if (strv_isempty((char**) value))
+                /* Case 2: the array is explicitly set to empty by the administrator */
+                printf("%13s %sdisabled by administrator%s\n", heading, ansi_highlight_red(), ansi_normal());
         else if (!field)
                 /* Case 2: we have values, but the field is NULL. This means that we're using the defaults.
```
Comment (AdrianVovk) [inline src/shared/user-record-show.c]: This is now case 3

**Review by AdrianVovk — APPROVED** [review]
LGTM code-wise

**Pull request merged**